pkg/providers/cloudstack/decoder/decoder.go

```diff
@@ -29,39 +30,57 @@ func ParseCloudStackSecret() (*CloudStackExecConfig, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to extract values from %s with ini: %v", EksacloudStackCloudConfigB64SecretKey, err)
 	}
-	section, err := cfg.GetSection("Global")
-	if err != nil {
-		return nil, fmt.Errorf("failed to extract section 'Global' from %s: %v", EksacloudStackCloudConfigB64SecretKey, err)
-	}
-	apiKey, err := section.GetKey("api-key")
-	if err != nil {
-		return nil, fmt.Errorf("failed to extract value of 'api-key' from %s: %v", EksacloudStackCloudConfigB64SecretKey, err)
-	}
-	secretKey, err := section.GetKey("secret-key")
-	if err != nil {
-		return nil, fmt.Errorf("failed to extract value of 'secret-key' from %s: %v", EksacloudStackCloudConfigB64SecretKey, err)
-	}
-	apiUrl, err := section.GetKey("api-url")
-	if err != nil {
-		return nil, fmt.Errorf("failed to extract value of 'api-url' from %s: %v", EksacloudStackCloudConfigB64SecretKey, err)
-	}
-	verifySsl, err := section.GetKey("verify-ssl")
-	verifySslValue := "true"
-	if err == nil {
-		verifySslValue = verifySsl.Value()
-		if _, err := strconv.ParseBool(verifySslValue); err != nil {
-			return nil, fmt.Errorf("'verify-ssl' has invalid boolean string %s: %v", verifySslValue, err)
+
+	cloudstackProfiles := []CloudStackProfileConfig{}
+	sections := cfg.Sections()
+	for _, section := range sections {
+		if section.Name() == "DEFAULT" {
+			continue
 		}
+
+		apiKey, err := section.GetKey("api-key")
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract value of 'api-key' from %s: %v", section.Name(), err)
+		}
+		secretKey, err := section.GetKey("secret-key")
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract value of 'secret-key' from %s: %v", EksacloudStackCloudConfigB64SecretKey, err)
+		}
+		apiUrl, err := section.GetKey("api-url")
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract value of 'api-url' from %s: %v", EksacloudStackCloudConfigB64SecretKey, err)
+		}
+		verifySslValue := "true"
+		if verifySsl, err := section.GetKey("verify-ssl"); err == nil {
+			verifySslValue = verifySsl.Value()
+			if _, err := strconv.ParseBool(verifySslValue); err != nil {
+				return nil, fmt.Errorf("'verify-ssl' has invalid boolean string %s: %v", verifySslValue, err)
+			}
+		}
+		cloudstackProfiles = append(cloudstackProfiles, CloudStackProfileConfig{
+			Name:          section.Name(),
```

Comment:
let's lowercase this to align with K8s secret naming conventions. You might also want to lowercase https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go#L180

Reply:
https://github.com/wongni/eks-anywhere/pull/5 will cover